components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -535,6 +536,13 @@ export const PortfolioAsset = (props: Props) => {
       }
       sendMessageToNftUiFrame(nftDetailsRef.current.contentWindow, command)
     }
+
+    // check if selectedAsset has an icon
+    if (selectedAsset && nftMetadata && stripERC20TokenImageURL(selectedAsset.logo) === '') {
```

Comment:
```suggestion
    if (selectedAsset && nftMetadata?.imageURL && stripERC20TokenImageURL(selectedAsset.logo) === '') {
```

Let's check if the metadata first, so that we aren't dispatching unneeded changes.

Reply:
Fixed 574e2cc547881632efbaec9513fe3689ab21576b